src/utils/dates.js

```diff
@@ -10,3 +10,30 @@ const ONE_DAY_IN_MILLISECONDS = 86400000;
 export function getFutureDate(offset) {
 	return new Date(Date.now() + offset * ONE_DAY_IN_MILLISECONDS);
 }
+
+/**
+ * Calculates the number of days between two JS Date objects.
+ * @example
+ * // Returns the number of days that have passed between the two dates.
+ * const startDate = new Date("2023-09-15");
+ * const endDate = new Date("2023-09-24");
+ * getDaysBetweenDates(startDate, endDate); // Output is 9 (days)
+ * @param {Date} startDate - The starting date.
+ * @param {Date} endDate - The ending date.
+ * @returns {number} - The number of days between the start and end dates.
+ */
+export function getDaysBetweenDates(startDate, endDate) {
+	try {
+		if (startDate instanceof Date && endDate instanceof Date) {
+			const days_between_in_milliseconds =
+				endDate.getTime() - startDate.getTime();
+
+			// Using Math.round() to ensure that fractional days are rounded to the nearest whole day
+			return Math.round(days_between_in_milliseconds / ONE_DAY_IN_MILLISECONDS);
```

Comment:
I've noticed for the items already on the list such as "Apples", we have `dateNextPurchased` < `dateLastPurchased`, while this is due to data being created prior to this issue and probably won't happen in the future for new items being added, does it make sense to add a ```return Math.abs(Math.round(days_between_in_milliseconds / ONE_DAY_IN_MILLISECONDS));``` just to make sure this `getDaysBetweenDates` always return a non-negative number representing a time interval?
![before check](https://github.com/the-collab-lab/tcl-63-smart-shopping-list/assets/24996005/2c9f19d1-9e73-4e85-9ec9-b9d1d1b6e4bd)
Notice previousEstimate = -28
![consolelog](https://github.com/the-collab-lab/tcl-63-smart-shopping-list/assets/24996005/5134f71e-cf39-4c1b-b153-9438d5bf793d)

Reply:
@shangguanwang Yep, you got that right! We added this `dateNextPurchased < dateLastPurchased` because we noticed those dates in the old items and wanted to prevent it from happening in the future. I believe what you're saying makes 100% sense, and I will go ahead and add the `Math.abs()`. Thank you for pointing that out!